Lib/test/libregrtest/worker.py

```diff
@@ -67,7 +68,8 @@ def worker_process(worker_json: StrJSON) -> NoReturn:
     runtests = RunTests.from_json(worker_json)
     test_name = runtests.tests[0]
     match_tests: FilterTuple | None = runtests.match_tests
-    json_file: JsonFile = runtests.json_file
+    json_file = runtests.json_file
+    assert isinstance(json_file, JsonFile)
```

Comment:
I dislike assert, it's removed by python -O. Why is this assertion needed? The attribute is declared as ``json_file: JsonFile | None``. Is it to remove the ``None`` case?

If mypy doesn't understand the logic, maybe a developer can have the same doubt. Maybe add a regular ``if json_file is None: raise ...``.

Or maybe, RunTests can have a sub-type which adds the json_file attribute: the class used by create_worker_process() / worker_process(). Would it please mypy?

Reply:
> Why is this assertion needed? The attribute is declared as `json_file: JsonFile | None`. Is it to remove the `None` case?

Yes, it's to remove the `None` case.

> If mypy doesn't understand the logic, maybe a developer can have the same doubt. Maybe add a regular `if json_file is None: raise ...`.

Sure. Should I do the same for all `assert`s being proposed in this PR?

> Or maybe, RunTests can have a sub-type which adds the json_file attribute: the class used by create_worker_process() / worker_process(). Would it please mypy?

Yes, that might also make mypy happier. Hard to tell without seeing a PR and running mypy on it, though :-)